### PR TITLE
Fix/activities

### PR DESCRIPTION
**Pull request by totorototo**
code clean-up

### CODE REVIEW

File: app/store/services/activities.js [hunk 1]
```diff
@@ -165,28 +165,27 @@ export const getRankings = (members = [], activities = {}) => {
...
-  let elevation = 0;
-  let duration = 0;
-
-  Object.keys(activities).forEach(id => {
-    distance += activities[id].distance;
-    elevation += activities[id].total_elevation_gain;
-    duration += activities[id].elapsed_time;
-  });
+  const sum = Object.keys(activities).reduce(
+    (accumulator, id) => ({
```
Comment (guipasmoi) [inline app/store/services/activities.js]: Object.entries(activities).reduce((accumulator, [id,activity])=>({.........}))

File: app/store/services/activities.js [hunk 2]
```diff
@@ -165,28 +165,27 @@ export const getRankings = (members = [], activities = {}) => {
...
-  let distance = 0;
-  let elevation = 0;
-  let duration = 0;
-
-  Object.keys(activities).forEach(id => {
-    distance += activities[id].distance;
-    elevation += activities[id].total_elevation_gain;
-    duration += activities[id].elapsed_time;
-  });
+  const sum = Object.keys(activities).reduce(
```
Comment (guipasmoi) [inline app/store/services/activities.js]: overalPerformance?

File: app/store/services/activities.js [hunk 3]
```diff
@@ -165,28 +165,27 @@ export const getRankings = (members = [], activities = {}) => {
...
+    (accumulator, id) => ({
+      distance: accumulator.distance + activities[id].distance,
+      elevation: accumulator.elevation + activities[id].total_elevation_gain,
+      duration: accumulator.duration + activities[id].elapsed_time
+    }),
+    { distance: 0, elevation: 0, duration: 0 }
+  );
 
-  const paceMeterPerSecond = duration > 0 ? distance / duration : 0;
+  const paceMeterPerSecond = sum.duration > 0 ? sum.distance / sum.duration : 0;
```
Comment (guipasmoi) [inline app/store/services/activities.js]: WTF? duration can be negative?

File: app/store/services/activities.js [hunk 4]
```diff
@@ -214,39 +213,38 @@ export const computePerformance = (activities = {}) => {
...
-    value: Object.keys(activities).length
-  });
-
   return {
     performance: {
-      details: performanceDetails,
+      details: [
+        {
+          name: "distance",
+          percent: Math.trunc(distanceHeuristic * 100 / performance),
```
Comment (guipasmoi) [inline app/store/services/activities.js]: don't format percent, it should be done at rendering

File: app/store/services/activities.js (hunk 3 above)
Comment (totorototo) [reply app/store/services/activities.js]: don't trust anyone! 🐰

File: app/store/services/activities.js (hunk 4 above)
Comment (totorototo) [reply app/store/services/activities.js]: you're right! it hurts!

File: app/store/services/activities.js (hunk 2 above)
Comment (totorototo) [reply app/store/services/activities.js]: better than sum, indeed.

File: app/store/services/activities.js (hunk 1 above)
Comment (totorototo) [reply app/store/services/activities.js]: something wrong.....

File: app/store/services/activities.js (hunk 4 above)
Comment (totorototo) [reply app/store/services/activities.js]: not sure about this one!

File: app/store/services/activities.js (hunk 1 above)
Comment (totorototo) [reply app/store/services/activities.js]: here: id = current value

File: app/store/services/activities.js (hunk 4 above)
Comment (guipasmoi) [reply app/store/services/activities.js]: i am  and i meant it for each

File: app/store/services/activities.js (hunk 1 above)
Comment (guipasmoi) [reply app/store/services/activities.js]: i used destructuring with array

File: app/store/services/activities.js (hunk 4 above)
Comment (totorototo) [reply app/store/services/activities.js]: what is heuristic value or unit for selectors or vues?
how could they know how to convert this from whatever it is to percent? 
this logic should stay in service! this is pure service stuff! This is the contract! 
Service return percent! Then selectors or view could do whatever they want / need with that!

Comment (guipasmoi) [reply app/store/services/activities.js]: 0.5 is 50% what you are doing is display logic

**Comment by totorototo** [comment]
this should be ok! Anyway, something must be done about cards and how their children could be displayed. Thoughts available for all kind of cards.

**Comment by guipasmoi** [comment]
you right :D

File: app/store/services/activities.js [hunk 5]
```diff
@@ -221,30 +221,30 @@ export const computePerformance = (activities = {}) => {
       details: [
         {
           name: "distance",
-          percent: Math.trunc(distanceHeuristic * 100 / performance),
-          value: Math.trunc(overallPerformance.distance / 1000),
+          percent: distanceHeuristic,
+          value: overallPerformance.distance,
```
Comment (guipasmoi) [inline app/store/services/activities.js]: was units wrong?

Comment (totorototo) [reply app/store/services/activities.js]: all of that should be handled by view - selector! 
psycho-rigid' styl'

**Comment by totorototo** [comment]
Got some issue with athlete details card. Got some with the brand new club feed card (new branch).

File: app/store/services/activities.js (hunk 5 above)
Comment (guipasmoi) [reply app/store/services/activities.js]: ok so why didnt you change units?

Comment (totorototo) [reply app/store/services/activities.js]: don't get your point! you want me to get ride of unit? 
ps: I was thinking about implementing a SI unit sytem to fix this!

File: app/store/services/activities.js [hunk 6]
```diff
@@ -234,18 +231,20 @@ export const computePerformance = (activities = {}) => {
...
+          value: overallPerformance.duration,
+          unit: "s"
         },
         {
           name: "pace",
           percent: paceHeuristic,
-          value: paceKilometerPerHour,
-          unit: "km/h"
+          value: paceMeterPerSecond,
+          unit: "m.s-1"
```
Comment (guipasmoi) [inline app/store/services/activities.js]: m/s

Comment (totorototo) [reply app/store/services/activities.js]: you should be kidding me!!!!!!!
m/s === m.s-1

File: app/store/services/activities.js [hunk 7]
```diff
@@ -234,18 +231,20 @@ export const computePerformance = (activities = {}) => {
...
         },
         {
           name: "runs count",
           percent: frequencyHeuristic,
-          value: Object.keys(activities).length
+          value: Object.keys(activities).length,
+          unit: ""
```
Comment (guipasmoi) [inline app/store/services/activities.js]: units: "number of runs" ????

**Review by guipasmoi — APPROVED** [review]
suggestion

File: app/store/services/activities.js (hunk 7 above)
Comment (totorototo) [reply app/store/services/activities.js]: not a si unit!

**Pull request merged**